src/sources/aws_s3/sqs.rs

```diff
@@ -513,18 +522,53 @@ impl IngestorProcess {
         let mut stream = lines.filter_map(move |line| {
             let mut log = LogEvent::from_bytes_legacy(&line).with_batch_notifier_option(&batch);
 
-            log.insert(event_path!("bucket"), bucket_name.clone());
-            log.insert(event_path!("object"), object_key.clone());
-            log.insert(event_path!("region"), aws_region.clone());
-            log.insert(log_schema().source_type_key(), Bytes::from("aws_s3"));
-            log.insert(log_schema().timestamp_key(), timestamp);
+            log_namespace.insert_source_metadata(
+                AwsS3Config::NAME,
+                &mut log,
+                path!("bucket"),
+                path!("bucket"),
+                bucket_name.clone(),
+            );
+            log_namespace.insert_source_metadata(
+                AwsS3Config::NAME,
+                &mut log,
+                path!("object"),
+                path!("object"),
+                object_key.clone(),
+            );
+            log_namespace.insert_source_metadata(
+                AwsS3Config::NAME,
+                &mut log,
+                path!("region"),
+                path!("region"),
+                aws_region.clone(),
+            );
 
             if let Some(metadata) = &metadata {
                 for (key, value) in metadata {
-                    log.insert(key.as_str(), value.clone());
+                    log_namespace.insert_source_metadata(
+                        AwsS3Config::NAME,
+                        &mut log,
+                        path!(key.as_str()),
```

Comment:
This path is different. Previously, it was using a full string path (which was parsed at runtime), and now it's creating a pre-compiled path of exactly 1 segment.

The easiest way to fix this is just remove the `path!` macro and continue passing just a string. This data comes from AWS, so we probably shouldn't be parsing it as a Vector path though, it's very possible for it to use invalid syntax, and is probably unreasonable to enforce that on S3 metadata, so we could consider making this a breaking change (but should document it).

Reply:
Ah, making sure I understand - this change could get a metadata key of `foo.bar.baz` and try and parse into into a path when it _should_ be `%aws_s3."foo.bar.baz"`?

Reply:
Interesting, and also inconsistent across components. Today the `kubernetes_logs` source will add labels as `.kubernetes.pod_labels."app.kubernetes.io/name"`...

Reply:
It feels more correct to me to not treat them as paths, but for this work I expect keeping the previous behavior makes sense.